home/export_assessments.py

```diff
@@ -0,0 +1,258 @@
+import copy
+import csv
+from dataclasses import asdict, astuple, dataclass, fields
+from math import ceil
+
+from django.http import HttpResponse  # type: ignore
+from openpyxl.styles import Border, Color, Font, NamedStyle, PatternFill, Side
+from openpyxl.utils import get_column_letter
+from openpyxl.workbook import Workbook
+from openpyxl.worksheet.worksheet import Worksheet
+from wagtail.blocks import (  # type: ignore
+    StreamValue,
+    StructValue,  # type: ignore
+)
+from wagtail.query import PageQuerySet  # type: ignore
+
+
+@dataclass
+class ExportRow:
+    title: str = ""
+    page_id: int = 0
+    tags: str = ""
+    slug: str = ""
+    locale: str = ""
+    high_result_page: str = ""
+    high_inflection: str = ""
+    medium_result_page: str = ""
+    medium_inflection: str = ""
+    low_result_page: str = ""
+    low_inflection: str = ""
+    generic_error: str = ""
+    question_count: int = 0
+    questions: str = ""
+    error: str = ""
+    answers: str = ""
+    score: str = ""
+
+    @classmethod
+    def headings(cls) -> list[str]:
+        return [f.name for f in fields(cls)]
+
+    def to_dict(self) -> dict[str, str | int]:
+        return asdict(self)
+
+    def to_tuple(self) -> tuple[str | int, ...]:
+        return astuple(self)
+
+
+# Define your AssessmentExporter class
+class AssessmentExporter:
+    rows: list[dict[str | int, str | int]]
+
+    def __init__(self, queryset: PageQuerySet):
+        self.rows = []
+        self.queryset = queryset
+
+    def perform_export(self) -> list[dict[str | int, str | int]]:
+        for item in self.queryset:
+            questions_data = self.get_questions(item.questions)
+            for i, question_data in enumerate(questions_data):
+
+                self.rows.append(
+                    {
+                        "title": item.title,
+                        "page_id": item.id,
```

Comment:
This isn't a page ID, this is an assessment. And I don't think we need the assessment ID in the export, we use the slug and locale as the unique identifier here.